tensorrt_llm/_torch/pyexecutor/py_executor.py

```
@@ -2821,6 +2821,22 @@ def _prepare_disagg_gen_transmission_complete(self, scheduled_batch):
 
+                # Prepend generation logits for first_gen_tokens if
+                # transferred from prefill.
+                disagg_params = getattr(req, 'py_disaggregated_params', None)
+                if (disagg_params is not None
+                        and getattr(disagg_params, 'first_gen_logits',
+                                    None) is not None):
+                    if beam_width != 1:
+                        raise ValueError(
+                            "first_gen_logits transfer currently assumes "
+                            "beam_width == 1; beam search is not supported "
+                            "with disaggregated logits propagation.")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)